### ISSUE TITLE
Provide a minimal working example for Spring Boot 3

### ISSUE THREAD

**Issue by eiswind** [spring-projects/spring-batch#4232]
Migrating to Spring Boot 3 causes us a lot of headache.

I tried to create a minimal (it is not working!) example like

```java
@Configuration
@EnableBatchProcessing
public class HelloBatch {


    @Bean
    public Step step(JobRepository jobRepository, PlatformTransactionManager tx) {
        return new StepBuilder( "step1", jobRepository)
                .tasklet((contribution, chunkContext) -> {
                    System.out.println("Hello, World!");
                    return RepeatStatus.FINISHED;
                }, tx).build();
    }

    @Bean
    public Job job(JobRepository jobRepository, Step step) {
        return new JobBuilder("job", jobRepository)
                .start(step)
                .build();
    }
}
```
Some working and tested examples would be very appreciated.

**Comment by fmbenhassine** [comment]
Thank you for opening this issue. 

> Provide a minimal working example for Spring Boot 3

The documentation/samples of Spring Batch do not cover the integration with Boot (not all Batch users are Boot users).

Please check the migration guide of Spring Boot here: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0.0-M5-Release-Notes#enablebatchprocessing-no-longer-required. `EnableBatchProcessing` is no longer required. I tried your sample with Spring Boot `3.0.0-RC2` (Spring Batch `5.0.0-RC2`) by removing `@EnableBatchProcessing` and the sample works as expected. 

I'm closing this issue for now, but if you need help to migrate your Spring Batch app to v5 please do no hesitate to add a comment here and we would be happy to help. You can find the v5 migration guide of Spring Batch in the wiki here: [Spring Batch 5.0 Migration Guide](https://github.com/spring-projects/spring-batch/wiki/Spring-Batch-5.0-Migration-Guide)

**Comment by eiswind** [comment]
@fmbenhassine Thanks for helping me out. Removing the annotation did help. But I still struggle with my tests, so I would be very happy if you could point my in the right direction. I created a test as described in the javadoc:

```java
@SpringBatchTest
@SpringJUnitConfig(HelloBatch.class)
public class HelloBatchTest {

    @Autowired
    private JobLauncherTestUtils jobLauncherTestUtils;
    
    @Test
    public void testHello(@Autowired Job job) throws Exception {

        jobLauncherTestUtils.setJob(job);

        var jobExecution =
                jobLauncherTestUtils.launchJob();

        assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());

    }
}
```

and this fails with
```
Error creating bean with name 'jobLauncherTestUtils': Unsatisfied dependency expressed through method 'setJobRepository' parameter 0: 
``` 

any tip whats wrong with this setup? (I always used to struggle with batch test setup somehow)

**Comment by eiswind** [comment]
Solved it.

Here adding @EnableBatchProcessing + @AutoConfigureJdbc made my day.

Thanks again for your patience.

**Comment by fmbenhassine** [comment]
This is a valid issue. Thank you for raising it! Please open a separate ticket for it and I will plan the fix for 5.0 GA. Thank you upfront!

**Comment by eiswind** [comment]
Thx. I created
https://github.com/spring-projects/spring-batch/issues/4233
